domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -884,63 +884,66 @@ public void saveBatch(String table, JsonArray entities, Handler<AsyncResult<Resu
     saveBatch(table, entities, DEFAULT_JSONB_FIELD_NAME, replyHandler);
   }
 
-  public void saveBatch(String table, JsonArray entities, String column, Handler<AsyncResult<ResultSet>> replyHandler) {
+  public void saveBatch(String table, JsonArray entities, String column,
+    Handler<AsyncResult<ResultSet>> replyHandler) {
+
+    client.getConnection(res -> {
+      if (res.failed()) {
+        replyHandler.handle(Future.failedFuture(res.cause()));
+        return;
+      }
+      saveBatch(res, table, entities, column, x -> {
+        res.result().close();
+        replyHandler.handle(x);
+      });
+    });
+  }
```

Comment:
a) This method consists mostly of duplicated code.
b) The res.failed() clause is not tested.

Please use the one liner we use for all new code:

        client.getConnection(conn -> saveBatch(conn, table, entities, column, closeAndHandleResult(conn, replyHandler)));

Move handling the res.failed() code into the saveBatch(res, ...) method.